system/jlib/jlog.hpp

```diff
@@ -1264,6 +1264,59 @@ interface jlib_decl IContextLogger : extends IInterface
     virtual const CRuntimeStatisticCollection & queryStats() const = 0;
 };
 
+// INTERFACE must be IContextLogger or derivative
+template <class INTERFACE>
+class CDefaultContextLogger : public CInterfaceOf<INTERFACE>
+{
+public:
+    virtual void CTXLOG(const char *format, ...) const  __attribute__((format(printf, 2, 3)))
+    {
+        va_list args;
+        va_start(args, format);
+        this->CTXLOGva(MCdebugInfo, unknownJob, NoLogMsgCode, format, args);
+        va_end(args);
+    }
+    virtual void mCTXLOG(const char *format, ...) const  __attribute__((format(printf, 2, 3)))
+    {
+        va_list args;
```

Comment:
minor: I would consider moving the bulk of this function into a shared implementation

` CContextLoggerHelper::mCTXLOG(IConectLogger * logger, const char * format, va_list & args);
`